modules/features/cu_hero_unit/cu_hero_unit.module

```diff
@@ -79,6 +79,9 @@ function cu_hero_unit_preprocess_entity(&$vars) {
     $classes_array[] = 'hero-text-' . render($vars['content']['field_hero_unit_text_color'][0]['#markup']);
     $classes_array[] = 'hero-background-' . render($vars['content']['field_hero_unit_bg_color'][0]['#markup']);
     $classes_array[] = 'hero-links-' . render($vars['content']['field_hero_unit_link_color'][0]['#markup']);
-    $vars['hero_classes'] = join(' ', $classes_array);  
+    if (!empty($vars['field_hero_unit_overlay'])) {
```

Comment:
No matter if I checked the box or not, this class still gets applied. I think you should switch it to check for === 1 instead of !empty()